packages/quantic/force-app/main/default/lwc/quanticRecommendationList/__tests__/quanticRecommendationList.test.js

```diff
@@ -0,0 +1,456 @@
+    beforeEach(() => {
+      mockSuccessfulHeadlessInitialization();
+      prepareHeadlessState();
+    });
+
+    describe('when there are no recommendations', () => {
+      it('should not display a recommendation list', async () => {
+        recommendationListState = {
+          isLoading: false,
+          recommendations: [],
+          error: null,
+          searchResponseId: null,
+        };
+        const element = createTestComponent();
+        await flushPromises();
+
+        const recommendationContainer = element.shadowRoot.querySelector(
+          selectors.gridContainer
+        );
+
+        expect(recommendationContainer).toBeNull();
+        const recommendationListItems = element.shadowRoot.querySelectorAll(
+          selectors.recommendationListItem
+        );
+        expect(recommendationListItems.length).toBe(0);
+      });
+    });
+
+    describe('when there are recommendations', () => {
+      describe('when the variant value is #grid', () => {
+        it('should properly display the recommendations with default options', async () => {
+          const element = createTestComponent({
+            ...defaultOptions,
+            variant: 'grid',
+          });
+          await flushPromises();
+
+          const recommendationContainer = element.shadowRoot.querySelector(
+            selectors.gridContainer
+          );
+          const recommendationItems = element.shadowRoot.querySelectorAll(
+            selectors.recommendationListItem
+          );
+
+          expect(recommendationContainer).not.toBeNull();
+          expect(recommendationItems.length).toBe(recommendationsArray.length);
+        });
+
+        describe('with a custom #recommendationsPerRow option', () => {
+          const customRecommendationsPerRow = 2;
+          it('should set the width of the recommendation items based on the number of #recommendationsPerRow value', async () => {
+            const expectedRecommendationWidth = `${100 / customRecommendationsPerRow}%`;
+            const element = createTestComponent({
+              ...defaultOptions,
+              variant: 'grid',
+              recommendationsPerRow: 2,
+            });
+            await flushPromises();
+
+            expect(element.style._values['--recommendationItemWidth']).toEqual(
+              expectedRecommendationWidth
+            );
+          });
+        });
+
+        it('should properly pass the custom #label to the heading', async () => {
+          const customLabel = 'Custom label';
+          const element = createTestComponent({
+            ...defaultOptions,
+            variant: 'grid',
+            label: customLabel,
+          });
+          await flushPromises();
+
+          const headingElement = element.shadowRoot.querySelector(
+            selectors.quanticHeading
+          );
+
+          expect(headingElement).not.toBeNull();
+          expect(headingElement.label).toEqual(customLabel);
```

Comment:
`toEqual` is to deep compare objects, for simple strings we use `toBe`
```suggestion
          expect(headingElement.label).toBe(customLabel);
```